### DISCUSSION TITLE
I don't receive The JOIN Accept from the GTW

### DISCUSSION THREAD

**Discussion by MEDOC-HUB** [Lora-net/LoRaMac-node#1289] [Q&A]
Hi,

I'm using the last release v4.6.0 (https://github.com/Lora-net/LoRaMac-node/commit/fe8247e2b84101fe701531a5f9ef14f035743af) with a NRF52832+Sx1262. I send every 15sec an JOIN Request  which know on the GTW but on the Rx Window,  the JOIN accept is never seen by the sx1262. You can see on the picture the different sequence about this.
![Screenshot-Sequence Join Failed v1](https://user-images.githubusercontent.com/73588461/154721463-8ce037d0-43d0-4f80-81a9-1073bfbc05b7.png)
Thanks for you help.

Best regards,
Mehdi GHARBI

**Comment by mluis1** [comment]
This is most certainly related to how the port to NRF52832 has been done.
Unfortunately we can not provide much advise on the porting to this specific platform, we did not had the chance to use this platform so far.
Maybe someone that has already done the porting to this platform can help you better. Under issue #631 there is a mention to 3 different ports. Maybe you can get inspiration from these.

Looking at your screenshots it looks like you have a timing issue. I would advise to review your timer functions.

In your screenshots the Rx1 window looks to happen after ~3.5 seconds. However the JoinAccept is expected to be received 5 seconds after the end of the transmission.
Something strange as well is that Rx2 window does not look to be opened. This should happen as there was no downlink on first window.

The best way to debug this kind of issues is to enable the compiling directive `USE_RADIO_DEBUG` under `.vscode/settings.json` file.

```json
{
...
        // Default channel plan for region CN470. Possible selections:
        // CHANNEL_PLAN_20MHZ_TYPE_A, CHANNEL_PLAN_20MHZ_TYPE_B, CHANNEL_PLAN_26MHZ_TYPE_A, CHANNEL_PLAN_26MHZ_TYPE_B
        "REGION_CN470_DEFAULT_CHANNEL_PLAN":"CHANNEL_PLAN_20MHZ_TYPE_A",

        // Enables radio debug pins
        "USE_RADIO_DEBUG":"ON"
}
```

Then connect the 2 MCU GPIO pins to an oscilloscope, you should be able to see when the radio is transmitting and when it is receiving.

If you have access to the gateway antenna port you could connect an RF splitter and then connect a cable to an oscilloscope (input impedance set to 50 Ohms) as well as to an antenna. This will show when the gateway transmits.

You should observe something like the following when JoinAccept is received (Tx: Green, Rx: Blue and GW-Tx: Yellow). Please note that the network server that I am currently using uses Rx2 window to send the JoinAccept (6 seconds after the end of the transmission)
![scope_7](https://user-images.githubusercontent.com/5763295/159225846-3681cd03-d84e-4b09-89ca-cb53d7077e73.png)

And the following when no JoinAccept is received
![scope_8](https://user-images.githubusercontent.com/5763295/159225946-6373fb15-e3a3-494e-bd21-ffe618c90144.png)

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.
